In [1]:
#imports 
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import urllib.parse
from glob import glob
import os
import ntpath 
import zipfile

#paths
#PPMI_zippath='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018.zip'
PPMI_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/'

In [2]:
def reading_csv_files(folder_path):
    # saving all the csv files in a dataframe
    import os
    import glob
    # glob.glob('*.csv') #find all the csv files in a pathname. 
    os.chdir(folder_path)
    csv_files = [i for i in glob.glob('*.csv')]

    # Reading each csv file and storing them in a dictionnary containing the file name and the dataframe
    dict_files={}
    files_names=[]
    for files in csv_files:
    #for i, files in enumerate(csv_files): 
        df=pd.read_csv(files,engine='python')
        basepath, filename=ntpath.split(files)
        files_names.append(filename)
        dict_files[filename]=df

    # Calculating the # of rows and features for each dataframe and storing them in a list of tuples
    shape_df=[]
    for i in range(0,len(dict_files.keys())):
        nrows, ncols=(len(dict_files[files_names[i]]),len(dict_files[files_names[i]].columns))
        shape_df.append((nrows,ncols))

    return (shape_df,files_names,dict_files)

In [3]:
def number_of_patients(df):
    '''return the number of patients in each csv files'''
    patients=len(list(set(df['PATNO'])))
    if 'EVENT_ID' in df.columns:
        events=len(list(set(df['EVENT_ID'])))
    else:
        events='no TS data'
    return (patients,events)

In [4]:
def features_selection(df, SEL):
    '''cleanier way to do features selection in each csv file'''
    return df_sel

In [5]:
def patients_selection(df, threshold):
    '''return a dataframe containing only the patients with number of visits > threshold'''
    return (patients_sel, df)

In [6]:
def patients_selection(df, threshold):
    '''return a dataframe containing only the patients with number of visits > threshold'''
    if 'PATNO' in df.columns: 
        visits_number_by_pat=df.groupby('PATNO').size().sort_values(ascending=False)
        mask_sel=visits_number_by_pat>=threshold
        patients_sel=list(mask_sel[mask_sel.values==True].index)
        df=df.loc[df['PATNO'].isin(patients_sel),:]
        df=df.sort_values('PATNO')
        return (patients_sel, df)
    else:
        print("There is no 'PATNO' column in the dataframe")

Inline-style: 
![1-Subject-Characteristics](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/1-subject-characteristics.png)
![2-Biospecimen](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/2-Biospecimen.png)
![3-Enrollment](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/3-Enrollment.png)
![4-Imaging](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/4-Medical.png)
![5-Medical-History](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/5-Imaging.png)
![6-Motor-Assessments](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/6-Motor-Assessments.png)
![7-Non-Motor-Assessments](https://github.com/AMDonati/parkinson-disease-project/blob/master/assets%20/7-Non-Motor-Assessments.png)

## 1-Subject-Characteristics

In [7]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/1-Subject-Characteristics/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### Patient_Status - actually useful for selecting only the PD patients in each table

In [8]:
PRODROMA=dict_files[files_names[0]]
print(number_of_patients(PRODROMA))
#print(list(set(PRODROMA['RECRUITMENT_CAT'])))
print(list(set(PRODROMA['ENROLL_CAT'])))
print(PRODROMA['ENROLL_CAT'].isnull().sum())
PRODROMA.head()
PD_cat=['PD','REGPD','GENPD']
mask=PRODROMA['ENROLL_CAT'].isin(PD_cat)
PD_patients=PRODROMA.loc[mask,'PATNO']
len(PD_patients)

(2135, 'no TS data')
[nan, 'PD', 'SWEDD', 'PRODROMA', 'REGPD', 'HC', 'GENUN', 'REGUN', 'GENPD']
331


876

### Screening__Demographics - useful
#### NOT TS DATA

In [9]:
SCREEN=dict_files[files_names[3]]
#SCREEN.head()
#list(set(SCREEN['EVENT_ID']))
#SCREEN.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME',
       'CONSNTDT', 'APPRDX', 'CURRENT_APPRDX', 'P3GRP', 'BIRTHDT', 'GENDER',
       'HISPLAT', 'RAINDALS', 'RAASIAN', 'RABLACK', 'RAHAWOPI', 'RAWHITE',
       'RANOS', 'PRJENRDT']
SCREEN_sel=SCREEN[SEL]
#list(set(SCREEN_sel['EVENT_ID']))
print(number_of_patients(SCREEN))
SCREEN_sel.head()

(2144, 1)


REC_ID  PATNO EVENT_ID PAG_NAME CONSNTDT  APPRDX  CURRENT_APPRDX  P3GRP  \
0  224392801   3400  CONSENT   SCREEN  06/2010     1.0             1.0    NaN   
1  224394301   3401  CONSENT   SCREEN  06/2010     2.0             2.0    NaN   
2  224398401   3402  CONSENT   SCREEN  06/2010     3.0             3.0    NaN   
3  224400201   3403  CONSENT   SCREEN  06/2010     1.0             1.0    NaN   
4  224718101   3404  CONSENT   SCREEN  06/2010     2.0             2.0    NaN   

   BIRTHDT  GENDER  HISPLAT  RAINDALS  RAASIAN  RABLACK  RAHAWOPI  RAWHITE  \
0   1971.0     0.0      0.0       0.0      0.0      0.0       0.0      1.0   
1   1954.0     1.0      0.0       0.0      0.0      0.0       0.0      1.0   
2   1964.0     2.0      0.0       0.0      0.0      0.0       0.0      1.0   
3   1941.0     2.0      0.0       0.0      0.0      0.0       0.0      1.0   
4   1954.0     0.0      0.0       0.0      0.0      0.0       0.0      1.0   

   RANOS PRJENRDT  
0    0.0  07/2010  
1    0.0  07/2010  
2    0.0  10/2011  
3    0.0  07/2010  
4    0.0  07/2010

## 2-Biospecimen

In [68]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/2-Biospecimen/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### Genetic_Results_Tests - useful: Feat-bio - 1116 patients
#### NOT TS DATA

In [69]:
MUTRSLT=dict_files[files_names[3]]
#MUTRSLT.head()
#MUTRSLT.columns

SEL=['REC_ID','PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'GENECAT', 'LRRKCD', 'MUTRSLT']
MUTRSLT_sel=MUTRSLT[SEL]
print(number_of_patients(MUTRSLT))
MUTRSLT_sel.head()

(1116, 1)


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  GENECAT  LRRKCD  MUTRSLT
0  426136501  41386      GMU  MUTRSLT  01/2014        1     1.0        1
1  445446201  50157      GMU  MUTRSLT  06/2014        1     1.0        1
2  422444501  40819      GMU  MUTRSLT  01/2014        1     1.0        1
3  423319501  40577      GMU  MUTRSLT  01/2014        1     1.0        1
4  493347201  41564      GMU  MUTRSLT  01/2015        1     1.0        1

### Pilot_Biospecimen_Analysis_Results - useful: Feat-bio
#### Understand if the cohort of patient IDs are different from the classic PPMI cohort.
#### NOT TS DATA.

In [12]:
### Pilot_Biospecimen_Analysis_Results - useful: Feat-bio
Biospecimen_Analysis=dict_files[files_names[6]]
#Biospecimen_Analysis.head()
#len(list(set(PILOTRSLT['PATNO'])))
#Biospecimen_Analysis.columns

SEL=['PATNO', 'DIAGNOSIS', 'CLINICAL_EVENT', 'TYPE', 'TESTNAME',
       'TESTVALUE','RUNDATE', 'PROJECTID']
print(number_of_patients(Biospecimen_Analysis))
Biospecimen_Analysis_sel=Biospecimen_Analysis[SEL]
Biospecimen_Analysis_sel.head()
#list(set(Biospecimen_Analysis_sel['DIAGNOSIS']))

(664, 'no TS data')


PATNO DIAGNOSIS CLINICAL_EVENT                 TYPE   TESTNAME TESTVALUE  \
0   3400        PD             BL  Cerebrospinal Fluid   Abeta 42       207   
1   3400        PD             BL  Cerebrospinal Fluid  p-Tau181P        14   
2   3400        PD             BL  Cerebrospinal Fluid  Total tau        10   
3   3400        PD             ST  Cerebrospinal Fluid   Abeta 42       231   
4   3400        PD             ST  Cerebrospinal Fluid  p-Tau181P        21   

      RUNDATE  PROJECTID  
0  2011-07-21        101  
1  2011-07-21        101  
2  2011-07-21        101  
3  2011-07-15        101  
4  2011-07-15        101

about Cerobrospinal fluid: https://en.wikipedia.org/wiki/Cerebrospinal_fluid

### Lumbar_Puncture_Sample_Collection - useful: Feat-bio - 1344 patients

In [72]:
LUMBAR=dict_files[files_names[9]]
#LUMBAR.head()
#LUMBAR.columns

### Only features useful I think: white cell blood count, red blood cell count, total glucose, total protein.

SEL=['REC_ID','PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT', 'FASTSTAT','PDMEDYN', 'PDMEDDT', 'PDMEDTM',
     'SMPDSCRD','RBCRSLT','WBCRSLT','TOPRRSLT','TGLCRSLT','SPFI','FLUORO']
print(number_of_patients(LUMBAR))
print(len(patients_selection(LUMBAR,6)[0]))
LUMBAR_sel=LUMBAR[SEL]
LUMBAR_sel.head()
#list(set(LUMBAR_sel['EVENT_ID']))

(1344, 20)
531


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  FASTSTAT  PDMEDYN PDMEDDT  \
0  232556301   3401       BL   LUMBAR  07/2010       NaN      0.0     NaN   
1  232568401   3403       BL   LUMBAR  07/2010       NaN      0.0     NaN   
2  233649201   3400       BL   LUMBAR  07/2010       NaN      0.0     NaN   
3  235052301   3404       BL   LUMBAR  07/2010       NaN      0.0     NaN   
4  236619201   3405       BL   LUMBAR  07/2010       NaN      0.0     NaN   

  PDMEDTM SMPDSCRD  RBCRSLT  WBCRSLT  TOPRRSLT  TGLCRSLT  SPFI  FLUORO  
0     NaN        0      0.0      1.0      20.0      63.0   0.0     0.0  
1     NaN        0      0.0      1.0      19.0      56.0   0.0     0.0  
2     NaN        0     76.0      1.0      75.0      55.0   0.0     0.0  
3     NaN        0      1.0      2.0      38.0      59.0   0.0     0.0  
4     NaN      NaN      NaN      NaN       NaN       NaN   NaN     NaN

## 3-Enrollment - useless?

In [14]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/3-Enrollment/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### 'Primary_Diagnosis.csv' - useful: Feat-Gen - 789 patients... (only PD patients?)

In [15]:
### 'Primary_Diagnosis.csv' - useful: Feat-Gen
PRIMDXPD=dict_files[files_names[8]]
#PRIMDXPD.head()
#PRIMDXPD.columns

SEL=['REC_ID', 'F_STATUS', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'PRIMDIAG', 'OTHNEURO']
print(number_of_patients(PRIMDXPD))
PRIMDXPD_sel=PRIMDXPD[SEL]
#list(set(PRIMDXPD['EVENT_ID']))
print(list(set(PRIMDXPD_sel['PRIMDIAG'])))
PRIMDXPD_sel.head()

(789, 15)
[nan, 97.0, 1.0, 4.0, 5.0, 7.0, 11.0, 15.0, 17.0]


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  PRIMDIAG OTHNEURO
0  226859501        V   3401       SC  PRIMDXPD  06/2010      17.0      NaN
1  232502501        V   3405       SC  PRIMDXPD  07/2010      17.0      NaN
2  226902401        V   3403       SC  PRIMDXPD  06/2010       1.0      NaN
3  232739901        V   3404       SC  PRIMDXPD  06/2010      17.0      NaN
4  232697801        V   3406       SC  PRIMDXPD  07/2010       1.0      NaN

### 5-Medical History

In [16]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/5-Medical-History/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### 'Diagnostic_Features.csv': useful: Feat-PD - 1670 patients.

In [17]:
DIAGFEAT=dict_files[files_names[1]]
DIAGFEAT.head()
#DIAGFEAT.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'DFSTROKE', 'DFRSKFCT', 'DFPRESNT', 'DFRPROG', 'DFSTATIC', 'DFHEMPRK',
       'DFAGESX', 'DFOTHCRS', 'DFCRSCM', 'DFRTREMP', 'DFRTREMA', 'DFPATREM',
       'DFOTHTRM', 'DFTREMCM', 'DFRIGIDP', 'DFRIGIDA', 'DFAXRIG', 'DFUNIRIG',
       'DFTONE', 'DFOTHRIG', 'DFRIGCM', 'DFBRADYP', 'DFBRADYA', 'DFAKINES',
       'DFBRPLUS', 'DFOTHABR', 'DFABRCM', 'DFPGDIST', 'DFGAIT', 'DFFREEZ',
       'DFFALLS', 'DFOTHPG', 'DFPGCM', 'DFPSYCH', 'DFCOGNIT', 'DFDYSTON',
       'DFCHOREA', 'DFMYOCLO', 'DFOTHHYP', 'DFHYPCM', 'DFHEMTRO', 'DFPSHYPO',
       'DFSEXDYS', 'DFURDYS', 'DFBWLDYS', 'DFOCULO', 'DFEYELID', 'DFNEURAB',
       'DFDOPRSP', 'DFRAPSPE', 'DFBULBAR', 'DFCTSCAN', 'DFMRI', 'DFATYP']
print(number_of_patients(DIAGFEAT))
DIAGFEAT_sel=DIAGFEAT[SEL]
DIAGFEAT_sel.head()
#list(set(DIAGFEAT_sel['EVENT_ID']))

(1670, 19)


REC_ID  PATNO EVENT_ID  PAG_NAME   INFODT  DFSTROKE  DFRSKFCT  DFPRESNT  \
0  358812401   3428      V04  DIAGFEAT  08/2012       0.0       0.0       0.0   
1  364251801   3433      V04  DIAGFEAT  09/2012       0.0       0.0       0.0   
2  366180201   3166      V04  DIAGFEAT  10/2012       0.0       0.0       0.0   
3  369126801   3162      V04  DIAGFEAT  10/2012       0.0       0.0       0.0   
4  362799201   3432       ST  DIAGFEAT  08/2012       0.0       0.0       0.0   

   DFRPROG  DFSTATIC   ...    DFBWLDYS  DFOCULO  DFEYELID DFNEURAB  DFDOPRSP  \
0      0.0       0.0   ...         0.0      0.0       0.0      0.0         0   
1      0.0       0.0   ...         0.0      0.0       0.0      0.0         N   
2      0.0       0.0   ...         0.0      0.0       0.0      0.0         N   
3      0.0       0.0   ...         0.0      0.0       0.0      0.0         N   
4      0.0       0.0   ...         0.0      0.0       0.0      0.0         N   

   DFRAPSPE  DFBULBAR  DFCTSCAN DFMRI  DFATYP  
0       0.0       0.0         0     0     0.0  
1       0.0       0.0         0     0     0.0  
2       0.0       0.0         N     0     0.0  
3       0.0       0.0         N     N     0.0  
4       0.0       0.0         0     0     0.0  

[5 rows x 59 columns]

### 'General_Medical_History.csv': useful: Feat-Gen - 2067 patients
#### NOT TS DATA...

In [18]:
MHXGEN=dict_files[files_names[3]]
#MHXGEN.head()
#MHXGEN.columns

SEL=['REC_ID', 'F_STATUS', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'MHCAT', 'MHROW', 'MHHX', 'MHACTRES', 'MHDIAGYR', 
     'PT_CODE', 'PT_NAME', 'HLT_NAME', 'SOCABBR1']
# I need probably to select less features here
print(number_of_patients(MHXGEN))
MHXGEN_sel=MHXGEN[SEL]
MHXGEN_sel.head()
#list(set(MHXGEN['EVENT_ID']))

(2067, 2)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  \
0  226698901        V   3400       SC   MHXGEN  06/2010   
1  226700701        V   3400       SC   MHXGEN  06/2010   
2  226701901        V   3400       SC   MHXGEN  06/2010   
3  226702601        V   3400       SC   MHXGEN  06/2010   
4  226870201        V   3401       SC   MHXGEN  06/2010   

                                             MHCAT  MHROW  MHHX  MHACTRES  \
0                                          1f. ENT      1   1.0       1.0   
1                         1l. Gynecologic/Urologic      1   1.0       1.0   
2  1p. Neurologic (other than disease under study)      1   0.0       NaN   
3  1r. Allergy / Immunologic (Note drug allergies)      2   1.0       1.0   
4                             1o. Hemato/lymphatic      1   0.0       NaN   

   MHDIAGYR     PT_CODE           PT_NAME  \
0    1987.0  10015090.0         Epistaxis   
1    2005.0  10013935.0     Dysmenorrhoea   
2       NaN         NaN               NaN   
3    1981.0  10020751.0  Hypersensitivity   
4       NaN         NaN               NaN   

                                HLT_NAME SOCABBR1  
0                    Nasal disorders NEC     Resp  
1  Menstruation and uterine bleeding NEC    Repro  
2                                    NaN      NaN  
3                Allergic conditions NEC    Immun  
4                                    NaN      NaN

### General_Physical_Exam: useful: Feat-Gen - 2047 patients

In [19]:
GENPHYEX=dict_files[files_names[5]]
#GENPHYEX.head()
#GENPHYEX.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'PESEQ', 'PECAT', 'ABNORM']
print(number_of_patients(GENPHYEX))
GENPHYEX_sel=GENPHYEX[SEL]
GENPHYEX_sel.head()
#list(set(GENPHYEX['EVENT_ID']))

(2047, 8)


REC_ID  PATNO EVENT_ID  PAG_NAME   INFODT  PESEQ  \
0  226890001   3402       SC  GENPHYEX  06/2010      3   
1  226890301   3402       SC  GENPHYEX  06/2010      6   
2  226890601   3402       SC  GENPHYEX  06/2010      9   
3  232699801   3406       SC  GENPHYEX  07/2010      8   
4  232700101   3406       SC  GENPHYEX  07/2010     11   

                                            PECAT  ABNORM  
0                                            Eyes     0.0  
1  Cardiovascular (including peripheral vascular)     0.0  
2                                    Neurological     0.0  
3                                 Musculoskeletal     0.0  
4           Other (Specify location and describe)     0.0

### 'General_Neurological_Exam.csv': useful: Feat-Gen - 2048 patients

In [20]:
PENEURO=dict_files[files_names[7]]
#PENEURO.head()
#PENEURO.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'MSRARSP', 'MSRACM', 'MSLARSP', 'MSLACM', 'MSRLRSP', 'MSRLCM',
       'MSLLRSP', 'MSLLCM', 'COFNRRSP', 'COFNRCM', 'COFNLRSP', 'COFNLCM',
       'COHSRRSP', 'COHSRCM', 'COHSLRSP', 'COHSLCM', 'SENRARSP', 'SENRACM',
       'SENLARSP', 'SENLACM', 'SENRLRSP', 'SENRLCM', 'SENLLRSP', 'SENLLCM',
       'RFLRARSP', 'RFLRACM', 'RFLLARSP', 'RFLLACM', 'RFLRLRSP', 'RFLRLCM',
       'RFLLLRSP', 'RFLLLCM', 'PLRRRSP', 'PLRRCM', 'PLRLRSP', 'PLRLCM']
print(number_of_patients(PENEURO))
PENEURO_sel=PENEURO[SEL]
PENEURO_sel.head()
#list(set(PENEURO['EVENT_ID']))

(2048, 15)


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  MSRARSP MSRACM  MSLARSP  \
0  232741801   3404       SC  PENEURO  06/2010        0    NaN        0   
1  232698601   3406       SC  PENEURO  07/2010        0    NaN        0   
2  226889601   3402       SC  PENEURO  06/2010        0    NaN        0   
3  226914401   3403       SC  PENEURO  06/2010        0    NaN        0   
4  226840201   3400       SC  PENEURO  06/2010        0    NaN        0   

  MSLACM  MSRLRSP   ...   RFLLARSP  RFLLACM RFLRLRSP  RFLRLCM RFLLLRSP  \
0    NaN        0   ...        1.0      NaN      1.0      NaN      1.0   
1    NaN        0   ...        2.0      NaN      2.0      NaN      2.0   
2    NaN        0   ...        2.0      NaN      2.0      NaN      2.0   
3    NaN        1   ...        3.0      NaN      3.0      NaN      3.0   
4    NaN        0   ...        2.0      NaN      2.0      NaN      2.0   

   RFLLLCM PLRRRSP  PLRRCM PLRLRSP  PLRLCM  
0      NaN     0.0     NaN     0.0     NaN  
1      NaN     0.0     NaN     0.0     NaN  
2      NaN     0.0     NaN     0.0     NaN  
3      NaN     0.0     NaN     0.0     NaN  
4      NaN     0.0     NaN     1.0     NaN  

[5 rows x 41 columns]

### 'Neurological_Exam_-_Cranial_Nerves.csv': useful: Feat-PD - 2049 patients

In [21]:
PENEURO2=dict_files[files_names[13]]
PENEURO2.head()
PENEURO2.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'CN1RSP','CN2RSP', 'CN346RSP','CN5RSP',
     'CN7RSP', 'CN8RSP', 'CN910RSP',
       'CN11RSP', 'CN12RSP']
print(number_of_patients(PENEURO2))
PENEURO2_sel=PENEURO2[SEL]
PENEURO2_sel.head()
#list(set(PENEURO['EVENT_ID']))

(2049, 15)


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  CN1RSP  CN2RSP  CN346RSP  \
0  232741701   3404       SC  PENEURO  06/2010     0.0     0.0       0.0   
1  232698501   3406       SC  PENEURO  07/2010     0.0     0.0       0.0   
2  226889501   3402       SC  PENEURO  06/2010     0.0     0.0       0.0   
3  226914301   3403       SC  PENEURO  06/2010     0.0     0.0       0.0   
4  226840101   3400       SC  PENEURO  06/2010     0.0     0.0       0.0   

   CN5RSP  CN7RSP  CN8RSP  CN910RSP  CN11RSP  CN12RSP  
0     0.0     0.0     0.0       0.0      0.0      0.0  
1     0.0     0.0     0.0       0.0      0.0      0.0  
2     0.0     0.0     0.0       0.0      0.0      0.0  
3     0.0     0.0     0.0       0.0      0.0      0.0  
4     0.0     0.0     0.0       0.0      0.0      0.0

### 'PD_Features.csv': useful: Feat-PD - caution: these are actually not timeseries but examination only done at screening (SC), BL & V04
### 1017 patients

In [22]:
PDFEAT=dict_files[files_names[14]]
#PDFEAT.head()
#PDFEAT.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT', 'SXMO',
       'SXYEAR', 'PDDXDT', 'DXTREMOR', 'DXRIGID', 'DXBRADY',
       'DXPOSINS', 'DXOTHSX', 'DXOTHCM', 'DOMSIDE']
print(number_of_patients(PDFEAT))
PDFEAT_sel=PDFEAT[SEL]
PDFEAT_sel.head()
#list(set(PDFEAT_sel['DXOTHCM'])) - see if we remove the comment feature or not.
#list(set(PDFEAT['EVENT_ID']))

(1017, 3)


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  SXMO  SXYEAR   PDDXDT  \
0  226923001   3403       SC   PDFEAT  06/2010   6.0    2009  03/2010   
1  226845501   3400       SC   PDFEAT  06/2010   3.0    2007  10/2009   
2  226895901   3402       SC   PDFEAT  06/2010   7.0    2008  03/2010   
3  232711801   3406       SC   PDFEAT  07/2010   6.0    2008  01/2010   
4  241672401   3407       SC   PDFEAT  08/2010   1.0    2010  02/2010   

  DXTREMOR DXRIGID DXBRADY DXPOSINS DXOTHSX DXOTHCM DOMSIDE  
0        0       1       0        0       0     NaN       2  
1        1       1       1        0       0     NaN       1  
2        1       0       1        0       0     NaN       2  
3        1       1       1        0       0     NaN       1  
4        1       1       1        0       0     NaN       1

### 'Vital_Signs.csv': useful: Feat-Gen
### 2047 patients

In [23]:
VITAL=dict_files[files_names[15]]
#VITAL.head()
#VITAL.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'WGTKG', 'HTCM', 'TEMPC', 'SYSSUP', 'DIASUP', 'HRSUP',
       'SYSSTND', 'DIASTND', 'HRSTND']
print(number_of_patients(VITAL))
VITAL_sel=VITAL[SEL]
VITAL_sel.head()

(2047, 21)


REC_ID  PATNO EVENT_ID PAG_NAME   INFODT  WGTKG  HTCM  TEMPC  SYSSUP  \
0  408779301  10362       SC    VITAL  08/2013    NaN   NaN   36.0   135.0   
1  605754701  10405       SC    VITAL  09/2014    NaN   NaN   36.9   137.0   
2  411032401  10496       SC    VITAL  06/2013    NaN   NaN   36.1   174.0   
3  409587001  10606       SC    VITAL  08/2013    NaN   NaN   36.6   170.0   
4  448228101  10662       SC    VITAL  07/2014    NaN   NaN   37.5   132.0   

   DIASUP  HRSUP  SYSSTND  DIASTND  HRSTND  
0    65.0   78.0    114.0     71.0    77.0  
1    81.0   64.0    150.0     80.0    73.0  
2    90.0   93.0    176.0     86.0    98.0  
3    92.0   51.0    170.0     61.0    58.0  
4    80.0   76.0    130.0     78.0    80.0

## 6-Motor-Assessments - all of them can be useful. 
#### But we might make the choice to not take all of them. 
#### For assessing the disease state (the labels), I think we should focus on the UPDRS1,3,&4 and take the other assessments as features

In [24]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/6-Motor-Assessments/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### 'MDS_UPDRS_Part_I__Patient_Questionnaire.csv' - 2011 patients

In [25]:
NUPDR1SP=dict_files[files_names[0]]
print(number_of_patients(NUPDR1SP))
NUPDR1SP.head()

(2011, 20)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  NUPSOURC  NP1SLPN  \
0  232593201        V   3400       SC  NUPDRS1P  07/2010       1.0      1.0   
1  230957601        V   3403       SC  NUPDRS1P  07/2010       1.0      0.0   
2  233164501        V   3400       BL  NUPDRS1P  07/2010       1.0      0.0   
3  234837801        V   3404       BL  NUPDRS1P  07/2010       1.0      1.0   
4  232529501        V   3401       BL  NUPDRS1P  07/2010       1.0      0.0   

   NP1SLPD  NP1PAIN  NP1URIN  NP1CNST  NP1LTHD  NP1FATG ORIG_ENTRY  \
0      1.0      3.0      1.0      2.0      1.0      3.0    07/2010   
1      1.0      1.0      0.0      3.0      0.0      2.0    07/2010   
2      1.0      2.0      1.0      1.0      1.0      4.0    07/2010   
3      1.0      0.0      1.0      0.0      0.0      1.0    07/2010   
4      0.0      0.0      0.0      0.0      0.0      0.0    07/2010   

             LAST_UPDATE QUERY SITE_APRV  
0  2010-07-13 16:42:58.0   NaN   07/2010  
1  2010-07-06 13:08:13.0   NaN   07/2010  
2  2010-07-15 17:09:39.0   NaN   07/2010  
3  2010-07-23 11:35:18.0   NaN   07/2010  
4  2010-07-13 13:36:12.0   NaN   07/2010

### 'MDS_UPDRS_Part_I.csv' - 2010 patients

In [26]:
NUPDRS1=dict_files[files_names[2]]
print(number_of_patients(NUPDRS1))
NUPDRS1.head()

(2010, 20)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NUPSOURC  NP1COG  \
0  231798601        V   3402       SC  NUPDRS1  07/2010       1.0     0.0   
1  233164301        V   3400       BL  NUPDRS1  07/2010       1.0     0.0   
2  230955201        V   3403       SC  NUPDRS1  07/2010       1.0     0.0   
3  234837501        V   3404       BL  NUPDRS1  07/2010       1.0     0.0   
4  232712901        V   3406       SC  NUPDRS1  07/2010       1.0     0.0   

   NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS ORIG_ENTRY  \
0      0.0      1.0      0.0      2.0     0.0    07/2010   
1      0.0      2.0      3.0      2.0     0.0    07/2010   
2      0.0      0.0      0.0      0.0     0.0    07/2010   
3      0.0      0.0      0.0      0.0     0.0    07/2010   
4      0.0      0.0      1.0      1.0     0.0    07/2010   

             LAST_UPDATE QUERY SITE_APRV  
0  2010-07-09 12:00:30.0   NaN   07/2010  
1  2010-07-15 17:04:33.0   NaN   07/2010  
2  2010-07-06 13:03:54.0   NaN   07/2010  
3  2010-07-23 11:30:12.0   NaN   07/2010  
4  2010-07-14 08:54:29.0   NaN   07/2010

### 'MDS_UPDRS_Part_II__Patient_Questionnaire.csv' - 2011 patients

In [27]:
NUPDRS2P=dict_files[files_names[11]]
print(number_of_patients(NUPDRS2P))
NUPDRS2P.head()

(2011, 20)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  NUPSOURC  NP2SPCH  \
0  232529901        V   3401       BL  NUPDRS2P  07/2010       1.0      0.0   
1  232713901        V   3406       SC  NUPDRS2P  07/2010       1.0      0.0   
2  231800401        V   3402       SC  NUPDRS2P  07/2010       1.0      0.0   
3  232595801        V   3400       SC  NUPDRS2P  07/2010       1.0      0.0   
4  230957201        V   3403       SC  NUPDRS2P  07/2010       1.0      0.0   

   NP2SALV  NP2SWAL    ...      NP2HOBB  NP2TURN  NP2TRMR  NP2RISE  NP2WALK  \
0      0.0      0.0    ...          0.0      0.0      0.0      0.0      0.0   
1      2.0      0.0    ...          2.0      0.0      3.0      0.0      0.0   
2      0.0      0.0    ...          0.0      0.0      1.0      0.0      0.0   
3      2.0      0.0    ...          1.0      1.0      1.0      1.0      1.0   
4      0.0      0.0    ...          1.0      1.0      0.0      1.0      1.0   

   NP2FREZ  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0      0.0     07/2010  2010-07-13 13:37:01.0    NaN    07/2010  
1      0.0     07/2010  2010-07-14 08:56:34.0    NaN    07/2010  
2      0.0     07/2010  2010-07-09 12:04:22.0    NaN    07/2010  
3      0.0     07/2010  2010-07-13 16:48:22.0    NaN    07/2010  
4      0.0     07/2010  2010-07-06 13:06:55.0    NaN    07/2010  

[5 rows x 24 columns]

### 'MDS_UPDRS_Part_III.csv' - 2010 patients

In [28]:
NUPDRS3=dict_files[files_names[7]]
print(number_of_patients(NUPDRS3))
NUPDRS3.head()

(2010, 20)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT CMEDTM EXAMTM  \
0  324141901        V   3506      V03  NUPDRS3  10/2011    NaN    NaN   
1  324180301        V   3419      V02  NUPDRS3  12/2011    NaN    NaN   
2  324389901        V   3816       BL  NUPDRS3  12/2011    NaN    NaN   
3  324396501        V   3762       BL  NUPDRS3  11/2011    NaN    NaN   
4  324399901        V   3765       BL  NUPDRS3  11/2011    NaN    NaN   

   NP3SPCH  NP3FACXP    ...      DYSKPRES  DYSKIRAT  NHY  \
0      1.0       1.0    ...           0.0       NaN  1.0   
1      0.0       0.0    ...           0.0       NaN  1.0   
2      0.0       0.0    ...           0.0       NaN  0.0   
3      0.0       1.0    ...           1.0       1.0  2.0   
4      0.0       0.0    ...           0.0       NaN  0.0   

   ANNUAL_TIME_BTW_DOSE_NUPDRS  ON_OFF_DOSE  PD_MED_USE  ORIG_ENTRY  \
0                          NaN          NaN           0     12/2011   
1                          NaN          NaN           0     12/2011   
2                          NaN          NaN           0     12/2011   
3                          NaN          NaN           0     12/2011   
4                          NaN          NaN           0     12/2011   

             LAST_UPDATE  QUERY  SITE_APRV  
0  2011-12-05 19:23:15.0    NaN    12/2011  
1  2011-12-06 07:25:10.0    NaN    12/2011  
2  2011-12-07 02:10:18.0    NaN    12/2011  
3  2011-12-07 02:34:41.0    NaN    12/2011  
4  2011-12-07 02:47:42.0    NaN    12/2011  

[5 rows x 51 columns]

### 'MDS_UPDRS_Part_IV.csv' - 860 patients

In [29]:
NUPDRS4=dict_files[files_names[5]]
print(number_of_patients(NUPDRS4))
NUPDRS4.head()

(860, 19)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NP4WDYSK  NP4DYSKI  \
0  267411701        V   3403       ST  NUPDRS4  01/2011       0.0       0.0   
1  280733601        V   3251      U01  NUPDRS4  02/2011       0.0       0.0   
2  304301401        V   3800       ST  NUPDRS4  07/2011       0.0       0.0   
3  303916001        V   3400      V04  NUPDRS4  07/2011       0.0       0.0   
4  306573401        V   3403      V04  NUPDRS4  07/2011       0.0       0.0   

   NP4OFF  NP4FLCTI  NP4FLCTX  NP4DYSTN ORIG_ENTRY            LAST_UPDATE  \
0     0.0       0.0       0.0       0.0    01/2011  2011-01-08 09:34:03.0   
1     0.0       0.0       0.0       0.0    03/2011  2011-03-16 12:19:26.0   
2     0.0       0.0       0.0       0.0    07/2011  2011-07-11 05:07:01.0   
3     0.0       0.0       0.0       0.0    07/2011  2011-07-07 08:06:50.0   
4     0.0       0.0       0.0       0.0    07/2011  2011-07-27 08:20:55.0   

  QUERY SITE_APRV  
0   NaN   01/2011  
1   NaN   03/2011  
2   NaN   07/2011  
3   NaN   07/2011  
4   NaN   07/2011

### 'PASE_-_Household_Activity.csv' - 1248 patients

In [30]:
PASE=dict_files[files_names[3]]
print(number_of_patients(PASE))
PASE.head()

(1248, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME  LTHSWRK  HVYHSWRK  HMREPR  \
0  456171801        V  10874       BL     PASE        2         1     1.0   
1  531597501        V  10874      V04     PASE        2         2     1.0   
2  585712501        V  10874      V06     PASE        2         2     1.0   
3  648651901        V  10874      V08     PASE        2         2     1.0   
4  416212301        V  12224       BL     PASE        2         2     1.0   

   LAWNWRK  OUTGARDN  CAREGVR  WRKVL  WRKVLHR  WRKVLACT ORIG_ENTRY  \
0        2       2.0      1.0    1.0      NaN       NaN    09/2014   
1        2       2.0      1.0    1.0      NaN       NaN    08/2015   
2        2       2.0      1.0    1.0      NaN       NaN    08/2016   
3        2       2.0      2.0    1.0      NaN       NaN    08/2017   
4        2       2.0      1.0    2.0      2.0       2.0    10/2013   

             LAST_UPDATE QUERY              SITE_APRV  
0  2014-09-11 09:49:22.0   NaN  2014-09-11 09:49:33.0  
1  2015-08-17 08:54:53.0   NaN  2015-08-17 08:55:04.0  
2  2016-08-24 11:52:50.0   NaN  2016-08-24 11:53:02.0  
3  2017-08-18 12:07:35.0   NaN  2017-08-18 12:11:04.0  
4  2013-10-30 12:23:09.0   NaN  2013-10-30 12:23:19.0

### 'PASE_-_Leisure_Time_Activity.csv' - 1248 patients

In [31]:
PASELT=dict_files[files_names[10]]
print(number_of_patients(PASELT))
PASELT.head()

(1248, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME  QUESTNO  ACTVOFT  \
0  456171201        V  10874       BL     PASE        1      3.0   
1  456171301        V  10874       BL     PASE        2      3.0   
2  456171401        V  10874       BL     PASE        3      3.0   
3  456171501        V  10874       BL     PASE        4      0.0   
4  456171601        V  10874       BL     PASE        5      0.0   

                ACTVSPEC  HRDAYFRQ ORIG_ENTRY            LAST_UPDATE QUERY  \
0  reading, computer, TV       3.0    09/2014  2014-09-11 09:49:22.0   NaN   
1                    NaN       2.0    09/2014  2014-09-11 09:49:22.0   NaN   
2              gardening       3.0    09/2014  2014-09-11 09:49:22.0   NaN   
3                    NaN       NaN    09/2014  2014-09-11 09:49:22.0   NaN   
4                    NaN       NaN    09/2014  2014-09-11 09:49:22.0   NaN   

               SITE_APRV  
0  2014-09-11 09:49:33.0  
1  2014-09-11 09:49:33.0  
2  2014-09-11 09:49:33.0  
3  2014-09-11 09:49:33.0  
4  2014-09-11 09:49:33.0

### 'Modified_Schwab_+_England_ADL.csv' - 1837 patients

In [32]:
MODSEADL=dict_files[files_names[6]]
print(number_of_patients(MODSEADL))
MODSEADL.head()

(1837, 20)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  MSEADLG ORIG_ENTRY  \
0  230954401        V   3403       SC  MODSEADL  06/2010     90.0    07/2010   
1  229763901        V   3402       SC  MODSEADL  06/2010     95.0    06/2010   
2  230939601        V   3400       SC  MODSEADL  06/2010     90.0    07/2010   
3  233166301        V   3400       BL  MODSEADL  07/2010     90.0    07/2010   
4  235635801        V   3406       SC  MODSEADL  07/2010     90.0    07/2010   

             LAST_UPDATE QUERY SITE_APRV  
0  2010-07-06 13:02:56.0   NaN   07/2010  
1  2010-06-30 13:49:58.0   NaN   06/2010  
2  2010-07-06 12:36:23.0   NaN   07/2010  
3  2010-07-15 17:20:22.0   NaN   07/2010  
4  2010-07-28 12:02:16.0   NaN   07/2010

## 7-Non-motor-assessments - all of them can be useful but we might need to choose. 
### Do a correlation analysis and ask doctors which ones are the more relevant. 

In [33]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/7-Non-motor-assessments/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### 'University_of_Pennsylvania_Smell_ID_Test.csv' - 1799 patients

In [34]:
UPSIT=dict_files[files_names[0]]
print(number_of_patients(UPSIT))
UPSIT.head()

(1799, 8)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  UPSITBK1  UPSITBK2  \
0  231754301        V   3403       BL    UPSIT  07/2010       4.0       2.0   
1  232551901        V   3401       BL    UPSIT  07/2010       9.0       9.0   
2  233828501        V   3400       BL    UPSIT  07/2010       6.0       8.0   
3  234847101        V   3404       BL    UPSIT  07/2010       8.0       6.0   
4  236093901        V   3405       BL    UPSIT  07/2010       9.0      10.0   

   UPSITBK3  UPSITBK4 NORMATIVE_SCORE COMM ORIG_ENTRY            LAST_UPDATE  \
0       1.0       2.0         Anosmia  NaN    07/2010  2010-07-09 09:53:39.0   
1       5.0       4.0        Hyposmia  NaN    07/2010  2010-07-13 14:23:31.0   
2       7.0       9.0        Hyposmia  NaN    07/2010  2010-07-20 07:37:29.0   
3       9.0      10.0        Hyposmia  NaN    07/2010  2010-07-23 12:15:13.0   
4      10.0      10.0       Normosmia  NaN    07/2010  2010-07-30 10:34:31.0   

  QUERY SITE_APRV  
0   NaN   07/2010  
1   NaN   07/2010  
2   NaN   07/2010  
3   NaN   07/2010  
4   NaN   07/2010

### 'Features_of_REM_Behavior_Disorder.csv' - only 80 patients...
#### NOT TS data.

In [35]:
REMBHVDS=dict_files[files_names[1]]
print(number_of_patients(REMBHVDS))
REMBHVDS.head()

(80, 1)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT REMONSDT  REMONEST  \
0  608680701        V  60048       SC  REMBHVDS  06/2014  01/1987       2.0   
1  608900501        V  60033       SC  REMBHVDS  12/2016  06/1996       3.0   
2  609531601        V  60065       SC  REMBHVDS  03/2014  01/2005       3.0   
3  609625701        V  60146       SC  REMBHVDS  12/2014  01/2007       3.0   
4  609524901        V  60059       SC  REMBHVDS  02/2014  12/2002       3.0   

   RBDDXDT  RBDDXEST    ...      ONBENZ  ONMLATON  ONSSRI  ONNORSRI  ONTRIADP  \
0  07/2007         1    ...         0.0       0.0     0.0       0.0       0.0   
1  05/2014         1    ...         0.0       1.0     1.0       1.0       0.0   
2  01/2010         1    ...         0.0       0.0     0.0       0.0       0.0   
3  04/2011         1    ...         0.0       0.0     0.0       0.0       0.0   
4  02/2011         1    ...         0.0       0.0     0.0       0.0       0.0   

   ONBTABLK  ORIG_ENTRY            LAST_UPDATE QUERY  SITE_APRV  
0       0.0     12/2016  2016-12-21 08:10:43.0   NaN    12/2016  
1       0.0     12/2016  2016-12-22 09:13:32.0   NaN    12/2016  
2       0.0     12/2016  2016-12-29 04:02:29.0   NaN    12/2016  
3       0.0     12/2016  2016-12-30 01:06:15.0   NaN    12/2016  
4       0.0     12/2016  2016-12-28 23:53:04.0   NaN    12/2016  

[5 rows x 21 columns]

### 'Epworth_Sleepiness_Scale.csv - 1343 patients.

In [36]:
EPWORTH=dict_files[files_names[2]]
print(number_of_patients(EPWORTH))
EPWORTH.head()

(1343, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  PTCGBOTH  ESS1  ESS2  \
0  234842801        V   3404       BL  EPWORTH  07/2010       1.0   1.0   1.0   
1  233819801        V   3400       BL  EPWORTH  07/2010       1.0   2.0   2.0   
2  232533701        V   3401       BL  EPWORTH  07/2010       1.0   1.0   1.0   
3  236092201        V   3405       BL  EPWORTH  07/2010       1.0   0.0   0.0   
4  231743201        V   3403       BL  EPWORTH  07/2010       1.0   3.0   3.0   

   ESS3  ESS4  ESS5  ESS6  ESS7  ESS8 ORIG_ENTRY            LAST_UPDATE QUERY  \
0   1.0   0.0   3.0   0.0   1.0   0.0    07/2010  2010-07-23 11:54:04.0   NaN   
1   1.0   3.0   3.0   0.0   1.0   0.0    07/2010  2010-07-20 07:17:01.0   NaN   
2   0.0   1.0   2.0   0.0   0.0   0.0    07/2010  2010-07-13 13:44:34.0   NaN   
3   0.0   2.0   0.0   0.0   0.0   0.0    07/2010  2010-07-30 10:24:35.0   NaN   
4   0.0   0.0   2.0   0.0   0.0   0.0    07/2010  2010-07-09 09:39:50.0   NaN   

  SITE_APRV  
0   07/2010  
1   07/2010  
2   07/2010  
3   07/2010  
4   07/2010

### 'Letter_-_Number_Sequencing__PD_.csv' - 1341 patients.

In [37]:
LNSPD=dict_files[files_names[3]]
print(number_of_patients(LNSPD))
LNSPD.head()

(1341, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  LNS1A  LNS1B  LNS1C  \
0  233165701        V   3400       BL    LNSPD  07/2010    1.0    1.0    1.0   
1  232532701        V   3401       BL    LNSPD  07/2010    1.0    1.0    1.0   
2  234840901        V   3404       BL    LNSPD  07/2010    1.0    1.0    1.0   
3  231742701        V   3403       BL    LNSPD  07/2010    1.0    1.0    1.0   
4  236091801        V   3405       BL    LNSPD  07/2010    1.0    1.0    1.0   

   LNS2A    ...      LNS7A  LNS7B  LNS7C  LNS_TOTRAW  AGE_ASSESS_LNS  DVS_LNS  \
0    1.0    ...        NaN    NaN    NaN        10.0            39.0      9.0   
1    1.0    ...        NaN    NaN    NaN        10.0            56.0     11.0   
2    1.0    ...        0.0    0.0    1.0        15.0            55.0     16.0   
3    1.0    ...        NaN    NaN    NaN        10.0            68.0     11.0   
4    1.0    ...        NaN    NaN    NaN         9.0            63.0     10.0   

   ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0     07/2010  2010-07-15 17:16:27.0    NaN    07/2010  
1     07/2010  2010-07-13 13:43:06.0    NaN    07/2010  
2     07/2010  2010-07-23 11:51:30.0    NaN    07/2010  
3     07/2010  2010-07-09 09:38:07.0    NaN    07/2010  
4     07/2010  2010-07-30 10:21:29.0    NaN    07/2010  

[5 rows x 34 columns]

### 'Olfactory_UPSIT.csv' - 4468 patients
#### NB: only take the 2 last columns as features (TOTAL_CORRECT, UPSIIT_PRCNTGE)

In [38]:
OLFACT=dict_files[files_names[4]]
#print(number_of_patients(OLFACT))
print(len(list(set(OLFACT['SUBJECT_ID']))))
OLFACT.head()

4668


SUBJECT_ID    PROTOCOL_ID COMPLT_DATE  SCENT_10_RESPONSE  \
0       90000  P-PPMI Cohort     04/2013                  2   
1       90001  P-PPMI Cohort     04/2013                  2   
2       90002  P-PPMI Cohort     03/2013                  2   
3       90003  P-PPMI Cohort     04/2013                  3   
4       90004  P-PPMI Cohort     04/2013                  1   

   SCENT_09_RESPONSE  SCENT_08_RESPONSE  SCENT_07_RESPONSE  SCENT_06_RESPONSE  \
0                  3                  2                  1                  2   
1                  3                  2                  1                  2   
2                  3                  2                  1                  2   
3                  1                  1                  4                  2   
4                  3                  2                  1                  2   

   SCENT_05_RESPONSE  SCENT_04_RESPONSE       ...        SCENT_38_CORRECT  \
0                  3                  4       ...                       1   
1                  3                  4       ...                       1   
2                  1                  4       ...                       1   
3                  4                  3       ...                       0   
4                  3                  4       ...                       1   

   SCENT_37_CORRECT  SCENT_36_CORRECT  SCENT_35_CORRECT  SCENT_34_CORRECT  \
0                 1                 0                 1                 1   
1                 0                 1                 1                 0   
2                 1                 1                 1                 0   
3                 0                 0                 0                 0   
4                 1                 0                 1                 1   

   SCENT_33_CORRECT  SCENT_32_CORRECT  SCENT_31_CORRECT  TOTAL_CORRECT  \
0                 1                 1                 1             38   
1                 1                 0                 1             30   
2                 1                 1                 1             28   
3                 0                 0                 0              8   
4                 1                 0                 1             32   

   UPSIIT_PRCNTGE  
0              98  
1              36  
2              36  
3               0  
4              75  

[5 rows x 85 columns]

### 'REM_Sleep_Disorder_Questionnaire.csv' - 1810 patients

In [39]:
REMSLEEP=dict_files[files_names[5]]
print(number_of_patients(REMSLEEP))
REMSLEEP.head()

(1810, 14)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  PTCGBOTH  DRMVIVID  \
0  232534301        V   3401       BL  REMSLEEP  07/2010       1.0       1.0   
1  233821601        V   3400       BL  REMSLEEP  07/2010       1.0       1.0   
2  234843201        V   3404       BL  REMSLEEP  07/2010       1.0       1.0   
3  231743401        V   3403       BL  REMSLEEP  07/2010       1.0       1.0   
4  236092401        V   3405       BL  REMSLEEP  07/2010       1.0       1.0   

   DRMAGRAC  DRMNOCTB    ...      NARCLPSY  DEPRS  EPILEPSY  BRNINFM  CNSOTH  \
0       0.0       0.0    ...           0.0    0.0       0.0      0.0     0.0   
1       1.0       0.0    ...           0.0    1.0       0.0      0.0     0.0   
2       0.0       0.0    ...           0.0    0.0       0.0      0.0     0.0   
3       1.0       0.0    ...           0.0    0.0       0.0      0.0     0.0   
4       1.0       0.0    ...           0.0    0.0       0.0      0.0     0.0   

   CNSOTHCM  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0       NaN     07/2010  2010-07-13 13:46:12.0    NaN    07/2010  
1       NaN     07/2010  2010-07-20 07:19:50.0    NaN    07/2010  
2       NaN     07/2010  2010-07-23 11:55:28.0    NaN    07/2010  
3       NaN     07/2010  2010-07-09 09:41:25.0    NaN    07/2010  
4       NaN     07/2010  2010-07-30 10:27:21.0    NaN    07/2010  

[5 rows x 33 columns]

### 'Cognitive_Assessments.csv' - 1172 patients

In [40]:
COGTIME=dict_files[files_names[6]]
print(number_of_patients(COGTIME))
COGTIME.head()

(1172, 11)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  HVLTRTTM  HVLTDRTM  \
0  517792401        V   3914      V06  COGTIME  04/2015  11:18:00  11:45:00   
1  518015001        V   3871      V06  COGTIME  04/2015       NaN       NaN   
2  518059301        V  41291       BL  COGTIME  04/2015  18:00:00  18:20:00   
3  520658401        V   4114      V06  COGTIME  05/2015  14:58:00  15:38:00   
4  520722501        V   3446      V08  COGTIME  05/2015  09:58:00  10:17:00   

   LNORNTTM     SFTTM     LNSTM     SDMTM  \
0       NaN       NaN       NaN       NaN   
1       NaN       NaN       NaN       NaN   
2  18:25:00  18:30:00  18:33:00  18:43:00   
3  15:12:00  15:18:00  15:26:00  15:32:00   
4  10:03:00  10:05:00  10:09:00  10:12:00   

                                                COMM ORIG_ENTRY  \
0  Only noted times for HVLT-R during this visit....    04/2015   
1  Not completed- test introduced only very recen...    04/2015   
2                                                NaN    04/2015   
3                                                NaN    05/2015   
4                                                NaN    05/2015   

             LAST_UPDATE QUERY SITE_APRV  
0  2015-04-29 03:01:42.0   NaN   04/2015  
1  2015-04-30 01:38:31.0   NaN   05/2015  
2  2015-04-30 06:26:32.0   NaN   04/2015  
3  2015-05-05 08:43:23.0   NaN   05/2015  
4  2015-05-05 12:00:25.0   NaN   05/2015

### 'SCOPA-AUT.csv' - 1345 patients

In [41]:
SCOPAAUT=dict_files[files_names[7]]
print(number_of_patients(SCOPAAUT))
SCOPAAUT.head()

(1345, 13)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  PTCGBOTH  SCAU1  \
0  233827401        V   3400       BL  SCOPAAUT  07/2010       1.0    0.0   
1  232551401        V   3401       BL  SCOPAAUT  07/2010       1.0    0.0   
2  234844601        V   3404       BL  SCOPAAUT  07/2010       1.0    0.0   
3  236093701        V   3405       BL  SCOPAAUT  07/2010       1.0    0.0   
4  231753901        V   3403       BL  SCOPAAUT  07/2010       1.0    0.0   

   SCAU2  SCAU3    ...      SCAU26B  SCAU26BT  SCAU26C   SCAU26CT  SCAU26D  \
0    1.0    0.0    ...          0.0       NaN      0.0        NaN      0.0   
1    0.0    0.0    ...          0.0       NaN      0.0        NaN      0.0   
2    0.0    0.0    ...          0.0       NaN      0.0        NaN      0.0   
3    0.0    0.0    ...          0.0       NaN      0.0        NaN      0.0   
4    0.0    0.0    ...          0.0       NaN      1.0  Enalapril      0.0   

   SCAU26DT  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0       NaN     07/2010  2012-01-08 08:13:13.0    NaN    07/2010  
1       NaN     07/2010  2012-01-08 08:13:14.0    NaN    07/2010  
2       NaN     07/2010  2012-01-08 08:13:14.0    NaN    07/2010  
3       NaN     07/2010  2012-01-08 08:13:14.0    NaN    07/2010  
4       NaN     07/2010  2012-01-08 08:13:14.0    NaN    07/2010  

[5 rows x 46 columns]

### Cognitive_Categorization.csv' - 1302 patients

In [42]:
COGCATG=dict_files[files_names[8]]
print(number_of_patients(COGCATG))
COGCATG.head()

(1302, 15)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  PTCGBOTH  COGDECLN  \
0  365358701        V   3574       ST  COGCATG  10/2012       1.0       0.0   
1  355330201        V   3179       BL  COGCATG  07/2012       3.0       0.0   
2  351869301        V   3178       BL  COGCATG  06/2012       1.0       1.0   
3  362798601        V   3432       ST  COGCATG  08/2012       1.0       0.0   
4  364318401        V   3407      V06  COGCATG  09/2012       1.0       0.0   

   FNCDTCOG  COGSTATE  COGDXCL  RVWNPSY ORIG_ENTRY            LAST_UPDATE  \
0       0.0       1.0      1.0      0.0    10/2012  2012-10-04 09:02:01.0   
1       0.0       1.0      1.0      1.0    07/2012  2012-07-23 17:22:03.0   
2       0.0       2.0      2.0      1.0    06/2012  2012-06-25 18:20:31.0   
3       0.0       1.0      1.0      0.0    09/2012  2012-09-18 13:44:55.0   
4       0.0       1.0      1.0      0.0    09/2012  2012-09-27 12:15:21.0   

  QUERY SITE_APRV  
0   NaN   10/2012  
1   NaN   07/2012  
2   NaN   06/2012  
3   NaN   09/2012  
4   NaN   09/2012

### 'Hopkins_Verbal_Learning_Test.csv' - 1341 patients

In [43]:
HVLT=dict_files[files_names[9]]
print(number_of_patients(HVLT))
HVLT.head()

(1341, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  HVLTRT1  HVLTRT2  \
0  231741801        V   3403       BL     HVLT  07/2010      6.0      7.0   
1  232531001        V   3401       BL     HVLT  07/2010      6.0     10.0   
2  233165101        V   3400       BL     HVLT  07/2010      8.0     10.0   
3  234838901        V   3404       BL     HVLT  07/2010      8.0     10.0   
4  236090801        V   3405       BL     HVLT  07/2010      6.0      9.0   

   HVLTRT3  HVLTRDLY    ...      comm  AGE_ASSESS_HVLT  DVT_TOTAL_RECALL  \
0      8.0       9.0    ...       NaN             68.0              44.0   
1     11.0      12.0    ...       NaN             56.0              49.0   
2     10.0      11.0    ...       NaN             39.0              49.0   
3     11.0      12.0    ...       NaN             55.0              54.0   
4     12.0      10.0    ...       NaN             63.0              51.0   

   DVT_DELAYED_RECALL DVT_RETENTION  DVT_RECOG_DISC_INDEX  ORIG_ENTRY  \
0                52.0          63.0                  47.0     07/2010   
1                61.0          62.0                  51.0     07/2010   
2                54.0          62.0                  58.0     07/2010   
3                61.0          62.0                  44.0     07/2010   
4                53.0          44.0                  52.0     07/2010   

             LAST_UPDATE  QUERY  SITE_APRV  
0  2012-06-03 09:05:07.0    NaN    07/2010  
1  2012-06-03 09:05:07.0    NaN    07/2010  
2  2012-06-03 09:05:07.0    NaN    07/2010  
3  2012-06-03 09:05:07.0    NaN    07/2010  
4  2012-06-03 09:05:08.0    NaN    07/2010  

[5 rows x 24 columns]

### 'Montreal_Cognitive_Assessment__MoCA_.csv' - 2054 patients

In [44]:
MOCA=dict_files[files_names[10]]
print(number_of_patients(MOCA))
MOCA.head()

(2054, 16)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  MCAALTTM  MCACUBE  \
0  226919901        V   3403       SC     MOCA  06/2010       1.0      1.0   
1  232509801        V   3401       SC     MOCA  07/2010       1.0      1.0   
2  232586001        V   3405       SC     MOCA  07/2010       1.0      0.0   
3  234494501        V   3406       SC     MOCA  07/2010       1.0      1.0   
4  226892001        V   3402       SC     MOCA  06/2010       1.0      1.0   

   MCACLCKC  MCACLCKN    ...      MCAMONTH  MCAYR  MCADAY  MCAPLACE  MCACITY  \
0       1.0       1.0    ...           1.0    1.0     1.0       1.0      1.0   
1       1.0       1.0    ...           1.0    1.0     1.0       1.0      1.0   
2       1.0       1.0    ...           1.0    1.0     1.0       1.0      1.0   
3       1.0       1.0    ...           1.0    1.0     1.0       1.0      1.0   
4       1.0       1.0    ...           1.0    1.0     1.0       1.0      1.0   

   MCATOT  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0    25.0     06/2010  2010-06-21 11:25:04.0    NaN    07/2010  
1    28.0     07/2010  2010-07-13 12:53:27.0    NaN    07/2010  
2    29.0     07/2010  2010-07-13 16:23:49.0    NaN    07/2010  
3    29.0     07/2010  2010-07-22 11:01:44.0    NaN    07/2010  
4    30.0     06/2010  2010-06-21 10:51:41.0    NaN    06/2010  

[5 rows x 38 columns]

### 'Semantic_Fluency.csv' - 1342 patients

In [45]:
SFT=dict_files[files_names[11]]
print(number_of_patients(SFT))
SFT.head()

(1342, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  VLTANIM  VLTVEG  \
0  233165501        V   3400       BL      SFT  07/2010       22      24   
1  232532101        V   3401       BL      SFT  07/2010       12      18   
2  234840401        V   3404       BL      SFT  07/2010       16      14   
3  231742501        V   3403       BL      SFT  07/2010       19       9   
4  236091601        V   3405       BL      SFT  07/2010       25      20   

   VLTFRUIT  AGE_ASSESS_SFTANIM  DVS_SFTANIM  DVT_SFTANIM ORIG_ENTRY  \
0        17                39.0         11.0         42.0    07/2010   
1        18                56.0          5.0         25.0    07/2010   
2        11                55.0          8.0         41.0    07/2010   
3        10                68.0         10.0         48.0    07/2010   
4        16                63.0         13.0         57.0    07/2010   

             LAST_UPDATE QUERY SITE_APRV  
0  2010-07-15 17:15:37.0   NaN   07/2010  
1  2010-07-13 13:42:06.0   NaN   07/2010  
2  2010-07-23 11:49:01.0   NaN   07/2010  
3  2010-07-09 09:37:11.0   NaN   07/2010  
4  2010-07-30 10:20:44.0   NaN   07/2010

### 'Geriatric_Depression_Scale__Short_.csv' - 1478 patients

In [46]:
GDSSHORT=dict_files[files_names[12]]
print(number_of_patients(GDSSHORT))
GDSSHORT.head()

(1478, 15)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  GDSSATIS  GDSDROPD  \
0  234843501        V   3404       BL  GDSSHORT  07/2010       1.0       0.0   
1  233822201        V   3400       BL  GDSSHORT  07/2010       1.0       0.0   
2  232534901        V   3401       BL  GDSSHORT  07/2010       1.0       0.0   
3  231744001        V   3403       BL  GDSSHORT  07/2010       1.0       0.0   
4  236092601        V   3405       BL  GDSSHORT  07/2010       1.0       0.0   

   GDSEMPTY  GDSBORED    ...      GDSMEMRY  GDSALIVE  GDSWRTLS  GDSENRGY  \
0       0.0       0.0    ...           0.0       1.0       0.0       0.0   
1       1.0       1.0    ...           0.0       0.0       0.0       0.0   
2       0.0       0.0    ...           0.0       1.0       0.0       1.0   
3       0.0       1.0    ...           0.0       1.0       0.0       0.0   
4       0.0       0.0    ...           0.0       1.0       0.0       1.0   

   GDSHOPLS  GDSBETER  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0       0.0       0.0     07/2010  2010-07-23 11:56:16.0    NaN    07/2010  
1       0.0       0.0     07/2010  2010-07-20 07:21:07.0    NaN    07/2010  
2       0.0       0.0     07/2010  2010-07-13 13:48:13.0    NaN    07/2010  
3       0.0       0.0     07/2010  2010-07-09 09:43:18.0    NaN    07/2010  
4       0.0       0.0     07/2010  2010-07-30 10:28:27.0    NaN    07/2010  

[5 rows x 25 columns]

### 'Symbol_Digit_Modalities.csv' - 1340 patients

In [47]:
SDM=dict_files[files_names[13]]
print(number_of_patients(SDM))
SDM.head()

(1340, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  SDMTOTAL  SDMTVRSN  \
0  231742901        V   3403       BL      SDM  07/2010      32.0       1.0   
1  232533101        V   3401       BL      SDM  07/2010      53.0       1.0   
2  233165901        V   3400       BL      SDM  07/2010      47.0       1.0   
3  234842101        V   3404       BL      SDM  07/2010      50.0       1.0   
4  236092001        V   3405       BL      SDM  07/2010      53.0       1.0   

   AGE_ASSESS_SDM  DVSD_SDM    DVT_SDM COMM ORIG_ENTRY            LAST_UPDATE  \
0            68.0    -1.000  40.000000  NaN    07/2010  2012-06-03 09:09:11.0   
1            56.0     0.625  56.250000  NaN    07/2010  2012-06-03 09:09:11.0   
2            39.0    -0.667  43.330002  NaN    07/2010  2012-06-03 09:09:11.0   
3            55.0     0.250  52.500000  NaN    07/2010  2012-06-03 09:09:11.0   
4            63.0     0.625  56.250000  NaN    07/2010  2012-06-03 09:09:11.0   

  QUERY SITE_APRV  
0   NaN   07/2010  
1   NaN   07/2010  
2   NaN   07/2010  
3   NaN   07/2010  
4   NaN   07/2010

### 'State-Trait_Anxiety_Inventory.csv' - 1474 patients

In [48]:
STAI=dict_files[files_names[14]]
print(number_of_patients(STAI))
STAI.head()

(1474, 15)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  STAIAD1  STAIAD2  \
0  234843901        V   3404       BL     STAI  07/2010      3.0      3.0   
1  233826001        V   3400       BL     STAI  07/2010      3.0      2.0   
2  232536101        V   3401       BL     STAI  07/2010      4.0      4.0   
3  236092801        V   3405       BL     STAI  07/2010      4.0      4.0   
4  231746001        V   3403       BL     STAI  07/2010      2.0      2.0   

   STAIAD3  STAIAD4    ...      STAIAD35  STAIAD36  STAIAD37  STAIAD38  \
0      1.0      1.0    ...           1.0       3.0       2.0       1.0   
1      1.0      1.0    ...           2.0       3.0       4.0       2.0   
2      1.0      1.0    ...           1.0       4.0       2.0       1.0   
3      1.0      1.0    ...           2.0       4.0       2.0       2.0   
4      2.0      2.0    ...           2.0       3.0       1.0       1.0   

   STAIAD39  STAIAD40  ORIG_ENTRY            LAST_UPDATE  QUERY  SITE_APRV  
0       3.0       2.0     07/2010  2010-07-23 11:58:39.0    NaN    07/2010  
1       3.0       4.0     07/2010  2010-07-20 07:28:30.0    NaN    07/2010  
2       4.0       1.0     07/2010  2010-07-13 13:52:33.0    NaN    07/2010  
3       4.0       1.0     07/2010  2010-07-30 10:30:33.0    NaN    07/2010  
4       3.0       2.0     07/2010  2010-07-09 09:47:13.0    NaN    07/2010  

[5 rows x 50 columns]

### 'Benton_Judgment_of_Line_Orientation.csv' - 1341 patients

In [49]:
BENTONOD=dict_files[files_names[15]]
print(number_of_patients(BENTONOD))
BENTONOD.head()

(1341, 13)


REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  BJLOT1  BJLOT2  \
0  233165301        V   3400       BL  BENTONOD  07/2010     1.0     NaN   
1  232531901        V   3401       BL  BENTONOD  07/2010     0.0     NaN   
2  234839901        V   3404       BL  BENTONOD  07/2010     1.0     NaN   
3  236091401        V   3405       BL  BENTONOD  07/2010     1.0     NaN   
4  231742101        V   3403       BL  BENTONOD  07/2010     1.0     NaN   

   BJLOT3  BJLOT4    ...      COMM  JLO_TOTRAW  JLO_TOTCALC  AGE_ASSESS_JLO  \
0     1.0     NaN    ...       NaN        13.0         26.0            39.0   
1     1.0     NaN    ...       NaN         9.0         18.0            56.0   
2     1.0     NaN    ...       NaN        15.0         30.0            55.0   
3     1.0     NaN    ...       NaN        14.0         28.0            63.0   
4     1.0     NaN    ...       NaN        13.0         26.0            68.0   

   DVS_JLO_MSSA  DVS_JLO_MSSAE  ORIG_ENTRY            LAST_UPDATE  QUERY  \
0          13.0          11.70     07/2010  2010-07-15 17:15:13.0    NaN   
1           8.0           5.97     07/2010  2010-07-13 13:41:35.0    NaN   
2          16.0          15.92     07/2010  2010-07-23 11:46:15.0    NaN   
3          14.0          12.80     07/2010  2010-07-30 10:20:02.0    NaN   
4          13.0          11.70     07/2010  2010-07-09 09:36:16.0    NaN   

   SITE_APRV  
0    07/2010  
1    07/2010  
2    07/2010  
3    07/2010  
4    07/2010  

[5 rows x 46 columns]

### 'QUIP_Current_Short.csv' - 1346 patients

In [50]:
QUIPCS=dict_files[files_names[16]]
print(number_of_patients(QUIPCS))
QUIPCS.head()

(1346, 13)


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  PTINBOTH  TMGAMBLE  \
0  232537101        V   3401       BL   QUIPCS  07/2010       1.0       0.0   
1  233826501        V   3400       BL   QUIPCS  07/2010       1.0       0.0   
2  234844401        V   3404       BL   QUIPCS  07/2010       1.0       0.0   
3  231746401        V   3403       BL   QUIPCS  07/2010       1.0       0.0   
4  236093001        V   3405       BL   QUIPCS  07/2010       1.0       0.0   

   CNTRLGMB  TMSEX    ...     CNTRLEAT  TMTORACT  TMTMTACT  TMTRWD  TMDISMED  \
0       0.0    0.0    ...          0.0       0.0       0.0     0.0         0   
1       0.0    0.0    ...          0.0       0.0       0.0     0.0         0   
2       0.0    0.0    ...          0.0       0.0       0.0     0.0         0   
3       0.0    0.0    ...          0.0       0.0       0.0     0.0         0   
4       0.0    0.0    ...          0.0       0.0       0.0     0.0         0   

   CNTRLDSM  ORIG_ENTRY            LAST_UPDATE QUERY SITE_APRV  
0         0     07/2010  2010-07-13 13:56:29.0   NaN   07/2010  
1         0     07/2010  2010-07-20 07:29:44.0   NaN   07/2010  
2         0     07/2010  2010-07-23 11:59:49.0   NaN   07/2010  
3         0     07/2010  2010-07-09 09:48:15.0   NaN   07/2010  
4         0     07/2010  2010-07-30 10:31:15.0   NaN   07/2010  

[5 rows x 24 columns]

# Datasets to investigate

### Blood_Chemistry__Hematology - Contains information about patient blood. 
### Ask doctors if really useful. 

In [51]:
COVANCE=dict_files[files_names[12]]
#COVANCE.head()

### IUSM_Catalog - DNK - most likely useless
#### seems ti be only metadata on the lab collections.
Look @ the PPMI doc to understand what is this table

In [52]:
IUSMCAT=dict_files[files_names[14]]
#IUSMCAT.head()

# Additionnal data that will be used later on (imaging data and medication data)

## 4-Imaging - dataset not to be used for the time being.

In [53]:
folder_path='/Users/alicemartin/02_DSR_Project/parkinson-disease-project/data/PPMI-final-dataset-382018/4-Imaging/'
files_names=reading_csv_files(folder_path)[1]
dict_files=reading_csv_files(folder_path)[2]

### FBB_Analysis_Data.csv

In [54]:
FBBANAL=dict_files[files_names[0]]
#FBBANAL.head()

### AV-133_Image_Metadata.csv

In [55]:
AV133META=dict_files[files_names[1]]
#AV133META.head()

### AV-133_Imaging.csv

In [56]:
AVIMAG=dict_files[files_names[2]]
#AVIMAG.head()

### DaTscan_Imaging.csv

In [57]:
DATSCAN=dict_files[files_names[3]]
#DATSCAN.head()

### SPECT_Scan_Information_Source_Document.csv

In [58]:
SPECTSCANINFO=dict_files[files_names[4]]
#SPECTSCANINFO.head()

### AV-133_SBR_Results.csv

In [59]:
AV133SBRRSLT=dict_files[files_names[5]]
#AV133SBRRSLT.head()

### Magnetic_Resonance_Imaging.csv

In [60]:
MRI=dict_files[files_names[6]]
#MRI.head()

### DTI_Regions_of_Interest

In [61]:
DTIROI=dict_files[files_names[7]]
#DTIROI.head()

### DaTSCAN_SPECT_Visual_Interpretation_Assessment.csv

In [62]:
FBBMETA=dict_files[files_names[8]]
#FBBMETA.head()

### 'FBB_Metadata.csv'

In [63]:
FBBMETA=dict_files[files_names[9]]
#FBBMETA.head()

### 'MRI_Imaging_Data_Transfer_Information_Source_Document.csv'

In [64]:
MRIINFO=dict_files[files_names[10]]
#MRIINFO.head()

## Medications data & misc data - to be used later on.

### Family_History_PD - most likely useful but not to be used for now.
#### Can be useful for detecting early PD

In [65]:
FAMHXPD=dict_files[files_names[2]]
#FAMHXPD.head()

### 'Prodromal_Diagnostic_Questionnaire.csv': useful: Feat-Gen (PRIMDIAG & PSLVL2)
#### NB: PSLV2 evaluates the proba of the patient being PD: might be useful for early PD. 
#### But not to be used in the main analysis as a feature (only for Prodromal - i.e 800 subjects)

In [66]:
PRODIAG=dict_files[files_names[0]]
#PRODIAG.head()
#PRODIAG.columns

SEL=['REC_ID', 'F_STATUS', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'PRIMDIAG',
       'OTHNEURO', 'PSLVL2']
print(number_of_patients(PRODIAG))
PRODIAG_sel=PRODIAG[SEL]
PRODIAG_sel.head()

(88, 6)


KeyError: "['REC_ID' 'F_STATUS' 'PAG_NAME' 'PRIMDIAG' 'OTHNEURO' 'PSLVL2'] not in index"

### 'Signature_Form.csv' - most likely useful: Meds - but not to be used for now.
#### Can be useful for the meds log

In [ ]:
SIG=dict_files[files_names[10]]
#SIG.head()

### Use of PD Medication: useful: Meds - but not to be used for now.

In [ ]:
PDMEDUSE=dict_files[files_names[4]]
#PDMEDUSE.head()

### 'Concomitant_Medications.csv': useful: Meds - But not to be used for now.

In [ ]:
CMED=dict_files[files_names[6]]
#CMED.head()

### 'Current_Medical_Conditions_Log.csv': useful: Meds - but not to be used for now. 

In [ ]:
### 'Current_Medical_Conditions_Log.csv': Meds
CURRCOND=dict_files[files_names[8]]
#CURRCOND.head()

### 'Adverse_Event_Log.csv': useful: Meds - but not to be used for now.
#### Measures the side effects of the meds taken for PD.

In [ ]:
LOG=dict_files[files_names[9]]
#LOG.head()

### 'Surgery_for_Parkinson_Disease.csv': useful: Meds - but not to be used for now.

In [ ]:
SURGPD=dict_files[files_names[12]]
#SURGPD.head()

## Datasets with two few data points (not enough patients)

### Skin_Biopsy - useful: Feat-bio (wound closure) - only 26 patients...

In [ ]:
SKBIO=dict_files[files_names[1]]
#SKBIO.head()
#SKBIO.columns

SEL=['REC_ID','PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT','SKBIOCMP','WOUNDCLS']
SKBIO_sel=SKBIO[SEL]
print(number_of_patients(SKBIO))
SKBIO_sel.head()
#list(set(SKBIO_sel['EVENT_ID']))

### IPSC_Labs: useful - Feat-bio: blood, liver disease, heparin(?), myeloma(?) stuff (linked to Med dose history)
### ONLY 137 patients...

In [ ]:
### ISPC_Labs: useful - Feat-bio
#### Select only history of liver disease and multiple myeloma (can be useful for treatments strategies)
IPSCLAB=dict_files[files_names[5]]
#IPSCLAB.head()
IPSCLAB.columns

SEL=['REC_ID','PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'PDMEDYN', 'PDMEDDT', 'PDMEDTM', 'WARFBEF', 'HEPBEF', 'HXLIVER',
       'HXMMYLMA', 'BLDLTH', 'LTHTM', 'LTHINVN', 'BLDSST', 'BLDCPT']
# we might be able to remove some features. 
print(number_of_patients(IPSCLAB))
IPSCLAB_sel=IPSCLAB[SEL]
IPSCLAB_sel.head()
#list(set(IPSCLAB_sel['EVENT_ID']))

### 'Clinical_Diagnosis_and_Management.csv': Feat-PD - only 73 patients...

In [ ]:
### 'Clinical_Diagnosis_and_Management.csv': Feat-PD
CLINDX=dict_files[files_names[2]]
CLINDX.head()
#CLINDX.columns

SEL=['REC_ID', 'PATNO', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'PSLVL', 'DCNOMTR', 'DCRTREM', 'DCRIGID', 'DCBRADY', 'DFPGDIST',
       'DFOTH', 'PRIMDIAG', 'OTHNEURO', 'CLDXCHNG', 'DXFDTIMG',
       'DXFCLSGN', 'DXFRSPMD', 'DXFNATHX', 'DXFOTH', 'DXFOTHCM', 'MNGCHNG',
       'MNGTRTSX', 'MNGTRIAL', 'MNGTRTOT', 'MNGDXTST', 'MNGNOTRT',
       'NEURWKDX']
print(number_of_patients(CLINDX))
CLINDX_sel=CLINDX[SEL]
CLINDX_sel.head()
#list(set(CLINDX_sel['EVENT_ID']))

### 'TAP-PD_OPDM_Assessment.csv' - only 173 patients...


In [ ]:
TAPOPDM=dict_files[files_names[8]]
print(number_of_patients(TAPOPDM))
TAPOPDM.head()

### 'TAP-PD_OPDM_Use_Questionnaire.csv' - only 32 patients

In [ ]:
TAPUSE=dict_files[files_names[1]]
print(number_of_patients(TAPUSE))
TAPUSE.head()

### 'Gait_Data___Arm_swing.csv' - not sure if we should consider this dataset. - only 62 patients...

In [ ]:
GAITDATA=dict_files[files_names[4]]
print(number_of_patients(GAITDATA))
GAITDATA.head()

### 'TAP-PD_Kinetics_Device_Testing.csv' - Idem for this one- not sure it is essential to consider it.
#### NO TS data. 
### Only 32 patients...

In [ ]:
KINETICS=dict_files[files_names[9]]
print(number_of_patients(KINETICS))
KINETICS.head()